src/app/components/pagination/pagination.component.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  JsonPipe,
+  NgForOf,
+  NgIf
+} from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output
+} from '@angular/core';
+
+type PaginationList = PageNumber[];
+type PageNumber = number;
+
+@Component({
+  selector: 'tmbd-pagination',
+  standalone: true,
+  templateUrl: './pagination.component.html',
+  imports: [
+    NgForOf,
+    NgIf,
+    JsonPipe
+  ],
+  styleUrls: ['./pagination.component.scss']
+})
+export class PaginationComponent {
+  @Input() public currentPage: number = 0;
+  @Input() public totalPages: number = 0;
+  @Output() public pageChange = new EventEmitter<number>();
+
+  private get pages(): PaginationList {
+    const pages = [];
```

Comment:
This function executes on every change detection cycle. Maybe it's better to define this variable after totalPages changed. You could create setter for totalPages and do it there.